components/engine_rocks/src/cf_names.rs

```diff
@@ -0,0 +1,10 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::engine::RocksEngine;
+use engine_traits::CFNamesExt;
+
+impl CFNamesExt for RocksEngine {
```

Comment:
I can not understand. Why we need to implement a single trait for 'cf_names()' ? Could it be member of Engine?

Reply:
It's because MiscExt contains a default method that needs to call `cf_names`, so MiscExt needs to be able to name a trait it lives on. It can't be `KvEngine` because KvEngine depends on MiscExt, not the other way around. Also I am generally moving away from KvEngine itself containing any methods.

It might be possible to just put `cf_names` on MiscExt. At the time I thought that wasn't possible, but I don't see why now.

Reply:
Ok, I see.